### PR TITLE
[PLAY-248] Alert Dialog Cleanup

### PR DESCRIPTION
**Pull request by Israel-Molestina**
#### Screens

Screen Shot 2022-08-30 at 4 33 58 PM
Screen Shot 2022-08-30 at 4 34 04 PM
Screen Shot 2022-08-30 at 4 34 10 PM
Screen Shot 2022-08-30 at 4 34 54 PM
Screen Shot 2022-08-30 at 4 34 59 PM
e695a.png)
Screen Shot 2022-08-30 at 4 35 10 PM
Screen Shot 2022-08-30 at 4 35 20 PM
Screen Shot 2022-08-30 at 4 35 24 PM

#### Breaking Changes

NO

#### Runway Ticket URL

[RUNWAY URL](https://nitro.powerhrg.com/runway/backlog_items/PLAY-248)

#### How to test this

Pull this branch and run yarn test-watch for jest test and cd playbook && ./test.sh for rails test. 

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] **SCREENSHOT** Please add a screen shot or two.
- [ ] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

### CODE REVIEW

**Review by nidaqg — APPROVED** [review]
Looks good!

**Comment by jasperfurniss** [comment]
Needs jest tests.

**Comment by jasperfurniss** [comment]
Once this has been finally reviewed tomorrow with @RexGluck, I'll take a look again at your PR and get it merged into the next release cycle.

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx [hunk 1]
```diff
@@ -10,53 +11,108 @@ const useDialog = (visible = false) => {
...
 
 const DialogStackedAlert = () => {
-  const [singleButtonOpen, toggleSingleButtonOpen] = useDialog()
-  const [stackedButtonOpen, toggleStackedButtonOpen] = useDialog()
-  const [singleLinkButtonOpen, toggleSingleLinkButtonOpen] = useDialog()
-  const [twoLinkButtonOpen, toggleTwoLinkButtonOpen] = useDialog()
+  const [defaultAlertOpened, toggleDefaultAlert] = useDialog()
+  const [cautionAlertOpened, toggleCautionAlert] = useDialog()
+  const [deleteAlertOpened, toggleDeleteAlert] = useDialog()
+  // const [singleButtonOpen, toggleSingleButtonOpen] = useDialog()
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx:17]: Let's remove all these comments.

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx [hunk 2]
```diff
@@ -10,53 +11,108 @@ const useDialog = (visible = false) => {
...
+      status: "delete",
+      text: "You are about to delete ...",
+      title: "Delete",
+      toggle: toggleDeleteAlert,
+      visible: deleteAlertOpened,
+      buttonOneText:"Yes, Delete",
+      buttonTwoText: "No, Cancel"
+    },
+    // {
+    //   size: 'sm',
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx:55]: Same. Remove these.

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx [hunk 3]
```diff
@@ -88,12 +144,37 @@ const DialogStackedAlert = () => {
...
+        >
+          <Button
+              fullWidth
+              onClick={dialog.toggle}
+              variant="secondary"
+          >
+            {dialog.buttonTwoText}
+          </Button>
+        </Dialog.Footer>
+        {/* <If condition={dialog.cancelledButton || dialog.confirmedButton}>
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx:177]: Remove.

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx (hunk 1 above)
Comment (Israel-Molestina) [reply playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx]: Done!

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx (hunk 2 above)
Comment (Israel-Molestina) [reply playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx]: Done

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx (hunk 3 above)
Comment (Israel-Molestina) [reply playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx]: Done

File: playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx [hunk 4]
```diff
@@ -37,7 +37,7 @@ type DialogProps = {
   opened: boolean,
   portalClassName?: string,
   shouldCloseOnOverlayClick: boolean,
-  size?: "sm" | "md" | "lg" | "content",
+  size?: "sm" | "md" | "lg" | 'status_size' | "content",
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx:40]:
```suggestion
  size?: "sm" | "md" | "lg" | "status_size" | "content",
```

This and a few other places using single ticks vs doubles.

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_status.jsx [hunk 5]
```diff
@@ -12,57 +12,70 @@ const useDialog = (visible = false) => {
...
-      buttonOneText:"No, Cancel",
-      buttonTwoText: "Yes, Action"
+      title: "Are you sure?",
+      toggle: toggleDefaultAlert,
+      visible: defaultAlertOpened,
+      buttonOneText:"Yes, Action",
+      buttonTwoText: "No, Cancel"
     },
     {
+      size: 'status_size',
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_status.jsx:34]:
```suggestion
      size: "status_size",
```

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_status.jsx [hunk 6]
```diff
@@ -12,57 +12,70 @@ const useDialog = (visible = false) => {
...
+  const [deleteAlertOpened, toggleDeleteAlert] = useDialog()
+  const [infoAlertOpened, toggleInfoAlert] = useDialog()
   const [successAlertOpened, toggleSuccessAlert] = useDialog()
   const [errorAlertOpened, toggleErrorAlert] = useDialog()
-  const [deleteAlertOpened, toggleDeleteAlert] = useDialog()
 
   const dialogs = [
     {
-      status: "info",
+      size: 'status_size',
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_status.jsx:24]:
```suggestion
      size: "status_size",
```

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_status.jsx [hunk 7]
```diff
@@ -12,57 +12,70 @@ const useDialog = (visible = false) => {
...
+      title: "Are you sure?",
       toggle: toggleCautionAlert,
       visible: cautionAlertOpened,
-      buttonOneText:"No, Cancel",
-      buttonTwoText: "Yes, Action"
+      buttonOneText:"Yes, Action",
+      buttonTwoText: "No, Cancel"
     },
     {
+      size: 'status_size',
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_status.jsx:44]:
```suggestion
      size: "status_size",
```

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_status.jsx [hunk 8]
```diff
@@ -12,57 +12,70 @@ const useDialog = (visible = false) => {
...
     },
     {
-      status: "error",
-      text: "Text explaining the error",
-      title: "Error Message",
-      toggle: toggleErrorAlert,
-      visible: errorAlertOpened,
-      buttonOneText:"No, Cancel",
-      buttonTwoText: "Ok, Thanks"
+      size: 'sm',
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_status.jsx:54]:
```suggestion
      size: "sm",
```

File: playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx (hunk 4 above)
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/_dialog.jsx:40]:
```suggestion
  size?: "sm" | "md" | "lg" | "status_size" | "content",
```

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx [hunk 9]
```diff
@@ -10,46 +9,40 @@ const useDialog = (visible = false) => {
...
-  const [twoLinkButtonOpen, toggleTwoLinkButtonOpen] = useDialog()
-
+  const [defaultAlertOpened, toggleDefaultAlert] = useDialog()
+  const [cautionAlertOpened, toggleCautionAlert] = useDialog()
+  const [deleteAlertOpened, toggleDeleteAlert] = useDialog()
 
   const dialogs = [
     {
-      status: "info",
+      size: 'sm',
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx:18]:
```suggestion
      size: "sm",
```

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx [hunk 10]
```diff
@@ -10,46 +9,40 @@ const useDialog = (visible = false) => {
...
-      cancelledButton: "Ok, Cancel",
-      toggle: toggleStackedButtonOpen,
-      visible: stackedButtonOpen,
+      toggle: toggleDefaultAlert,
+      visible: defaultAlertOpened,
+      buttonOneText:"Yes, Action",
+      buttonTwoText: "No, Cancel"
     },
     {
+      size: 'sm',
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx:28]:
```suggestion
      size: "sm",
```

File: playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx [hunk 11]
```diff
@@ -10,46 +9,40 @@ const useDialog = (visible = false) => {
...
     },
     {
-      status: "success",
-      text: "Text explaining what is successful",
-      title: "Success",
-      toggle: toggleTwoLinkButtonOpen,
-      visible: twoLinkButtonOpen,
-      linkConfirmedButton:"Ok",
-      linkCancelledButton: "Cancel"
+      size: 'sm',
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_dialog/docs/_dialog_stacked_alert.jsx:38]:
```suggestion
      size: "sm",
```

**Review by thestephenmarshall — CHANGES_REQUESTED** [review]
Still a lot of suggestions to commit here @Israel-Molestina

**Review by thestephenmarshall — APPROVED** [review]

**Pull request merged**